src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -454,11 +454,11 @@ private function resolveServices($value)
             $value = array_map(array($this, 'resolveServices'), $value);
         } elseif (is_string($value) &&  0 === strpos($value, '@=')) {
             return new Expression(substr($value, 2));
-        } elseif (is_string($value) &&  0 === strpos($value, '@')) {
-            if (0 === strpos($value, '@@')) {
+        } elseif (is_string($value) && $value[0] && '@' === $value[0]) {
+            if ('@' === $value[1]) {
                 $value = substr($value, 1);
                 $invalidBehavior = null;
```

Comment:
Rather than using ``$invalidBehavior = null`` to decide later that it is not a reference (and which will still remove the ``=`` at the end of the escaped string, which is a bug), I would rather use `` return substr($value, 1);`` here, to split things outside the handling of references